### PR TITLE
CIRC-699: Use partition to fetch loans for requests.

### PR DESCRIPTION
**Pull request by bohdan-suprun**
https://issues.folio.org/browse/CIRC-699 - Bugfest: 500 Error when filtering Requests by Type = Page

## Approach
Use `FindWithMultipleCqlIndexValues` to fetch open loans for items, when fetching multiple requests in order to omit `414 Uri too long` error.

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/test/java/api/requests/RequestsAPICreationTests.java [hunk 1]
```diff
@@ -1965,6 +1971,35 @@ public void cannotCreateRequestForClaimedReturnedItem(String requestType) {
...
+    UUID pickupServicePointId = servicePointsFixture.cd1().getId();
+    final List<IndividualResource> createdRequests =
+      IntStream.range(0, 100).mapToObj(notUsed -> requestsFixture.place(
+        new RequestBuilder()
+          .open()
+          .page()
+          .forItem(itemsFixture.basedUponSmallAngryPlanet())
+          .by(usersFixture.charlotte())
+          .fulfilToHoldShelf()
+          .withPickupServicePointId(pickupServicePointId)))
```
Comment (marcjohnson-kint) [inline src/test/java/api/requests/RequestsAPICreationTests.java]: I might extract the placing of the request into a method, to reduce the noise in this test.

File: src/test/java/api/requests/RequestsAPICreationTests.java [hunk 2]
```diff
@@ -1965,6 +1971,35 @@ public void cannotCreateRequestForClaimedReturnedItem(String requestType) {
     )));
   }
 
+  @Test
+  public void canFetchHundredRequests() {
+    UUID pickupServicePointId = servicePointsFixture.cd1().getId();
+    final List<IndividualResource> createdRequests =
```
Comment (marcjohnson-kint) [inline src/test/java/api/requests/RequestsAPICreationTests.java]: Personally, I would have a blank line between these two lines, to separate the multiple line initialisation out, and hopefully make it easier to read.

File: src/main/java/org/folio/circulation/domain/LoanRepository.java [hunk 3]
```diff
@@ -312,11 +314,10 @@ private static void keepPatronGroupIdAtCheckoutProperties(Loan loan, JsonObject
...
     }
 
-    final Result<CqlQuery> statusQuery = getStatusCQLQuery("Open");
-    final Result<CqlQuery> itemIdQuery = exactMatchAny(ITEM_ID, itemsToFetchLoansFor);
+    final FindWithMultipleCqlIndexValues<Loan> fetcher =
+      findWithMultipleCqlIndexValues(loansStorageClient, RECORDS_PROPERTY_NAME, Loan::from);
 
-    return queryLoanStorage(statusQuery.combine(
-        itemIdQuery, CqlQuery::and), PageLimit.limit(requests.size()))
+    return fetcher.findByIdIndexAndQuery(itemsToFetchLoansFor, ITEM_ID, getStatusCQLQuery("Open"))
```
Comment (marcjohnson-kint) [inline src/main/java/org/folio/circulation/domain/LoanRepository.java]: Was this the issue, that it was using an older way of fetching the loans?

File: src/test/java/api/requests/RequestsAPICreationTests.java [hunk 4]
```diff
@@ -1965,6 +1971,35 @@ public void cannotCreateRequestForClaimedReturnedItem(String requestType) {
...
+          .page()
+          .forItem(itemsFixture.basedUponSmallAngryPlanet())
+          .by(usersFixture.charlotte())
+          .fulfilToHoldShelf()
+          .withPickupServicePointId(pickupServicePointId)))
+        .collect(Collectors.toList());
+
+    final Map<String, JsonObject> foundRequests =
+      requestsFixture.getRequests(CqlQuery.exactMatch("requestType", "Page"),
+        Limit.limit(100), Offset.noOffset())
```
Comment (marcjohnson-kint) [inline src/test/java/api/requests/RequestsAPICreationTests.java]: I might statically import those methods to noise the distraction in this line.

File: src/main/java/org/folio/circulation/domain/LoanRepository.java (hunk 3 above)
Comment (bohdan-suprun) [reply src/main/java/org/folio/circulation/domain/LoanRepository.java]: Yes, it was. Here we use the old way of fetching multiple elements.

**Comment by marcjohnson-kint** [comment]
@bohdan-suprun When this is merged, please can you issue another pull request for this, to the b18.0 branch so it can be included in a bug fix release.

**Pull request merged**